rpcs3/rpcs3qt/pad_settings_dialog.cpp

```diff
@@ -1626,69 +1626,7 @@ void pad_settings_dialog::HandleDeviceClassChange(u32 class_id) const
 
 	for (const input::product_info& product : input::get_products_by_class(class_id))
 	{
-		switch (product.type)
-		{
-		case input::product_type::playstation_3_controller:
-		{
-			ui->chooseProduct->addItem(tr("PS3 Controller", "PlayStation 3 Controller"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::dance_dance_revolution_mat:
-		{
-			ui->chooseProduct->addItem(tr("Dance Dance Revolution", "Dance Dance Revolution Mat"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::dj_hero_turntable:
-		{
-			ui->chooseProduct->addItem(tr("DJ Hero Turntable", "DJ Hero Turntable"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::harmonix_rockband_drum_kit:
-		{
-			ui->chooseProduct->addItem(tr("Rock Band", "Harmonix Rock Band Drum Kit"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::harmonix_rockband_drum_kit_2:
-		{
-			ui->chooseProduct->addItem(tr("Rock Band Pro", "Harmonix Rock Band Pro-Drum Kit"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::harmonix_rockband_guitar:
-		{
-			ui->chooseProduct->addItem(tr("Rock Band", "Harmonix Rock Band Guitar"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::red_octane_gh_drum_kit:
-		{
-			ui->chooseProduct->addItem(tr("Guitar Hero", "RedOctane Guitar Hero Drum Kit"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::red_octane_gh_guitar:
-		{
-			ui->chooseProduct->addItem(tr("Guitar Hero", "RedOctane Guitar Hero Guitar"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::rock_revolution_drum_kit:
-		{
-			ui->chooseProduct->addItem(tr("Rock Revolution", "Rock Revolution Drum Controller"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::ps_move_navigation:
-		{
-			ui->chooseProduct->addItem(tr("PS Move Navigation", "PS Move Navigation Controller"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::ride_skateboard:
-		{
-			ui->chooseProduct->addItem(tr("RIDE Skateboard", "Tony Hawk RIDE Skateboard Controller"), static_cast<int>(product.type));
-			break;
-		}
-		case input::product_type::guncon_3:
-		{
-			ui->chooseProduct->addItem(tr("GunCon 3", "GunCon 3 Controller"), static_cast<int>(product.type));
-			break;
-		}
-		}
+		ui->chooseProduct->addItem(tr(product.name, product.description), static_cast<int>(product.type));
```

Comment:
Why did you change this?
Please revert this

Reply:
Why? This way the names are stored in the same map and new devices will not need changes in so many files.

Reply:
These are user facing strings that will be exported to linguist.
This has nothing to do with the product info struct

Reply:
So your code literally breaks the localization anyway.